test/server/cards/02_SHD/units/SlaversFreighter.spec.ts

```diff
@@ -0,0 +1,47 @@
+describe('Slaver\'s Freighter', function() {
+    integration(function(contextRef) {
+        describe('Slaver\'s Freighter\'s ability', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['slavers-freighter'],
+                        groundArena: [{ card: 'cell-block-guard', exhausted: true }, { card: 'count-dooku#darth-tyranus', exhausted: true }],
+                    },
+                    player2: {
+                        groundArena: [{ card: 'wampa', upgrades: ['fallen-lightsaber', 'electrostaff'], exhausted: true }],
+                        spaceArena: [{ card: 'green-squadron-awing', upgrades: ['resilient'] }]
+                    }
+                });
+            });
+
+            it('should allow a friendly unit with power less than or equal to the number of upgrades on enemy units to be readied.', function () {
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.slaversFreighter);
+
+                expect(context.player1).toHavePassAbilityButton();
+                expect(context.player1).toBeAbleToSelectExactly([context.cellBlockGuard, context.greenSquadronAwing]);
```

Comment:
Could we bump the number of upgrades to 5 and confirm that slaver's freighter can target itself. Also, after that could we replace Dooku with a 6-cost just to confirm that the upgrade count cutoff is exactly correct

Reply:
I just checked the wording of the card and it says ready another unit, so shouldn't be able to target itself, but your point still stands, I should test this with the higher number of upgrades, I will update

Reply:
Ahh good catch, I missed that. Thanks